modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -307,6 +307,69 @@ public CodegenModel fromModel(String name, Schema model) {
+            }
+
+            Map<String, String> refsMapping = Optional.ofNullable(model.getDiscriminator())
+                    .map(Discriminator::getMapping).orElse(Collections.emptyMap());
+
+            // Reverse mapped references to use as baseName for anyOf, but different keys may point to the same $ref.
+            // Thus, we group them by the value
+            Map<String, List<String>> mappedNamesByRef = refsMapping.entrySet().stream()
+                    .collect(Collectors.groupingBy(Map.Entry::getValue,
+                            Collectors.mapping(Map.Entry::getKey, Collectors.toList())
+                    ));
+
+            for (int i = 0; i < newAnyOfs.size(); i++) {
+                CodegenProperty anyOf = newAnyOfs.get(i);
+                Schema schema = schemas.get(i);
+
+                if (mappedNamesByRef.containsKey(schema.get$ref())) {
+                    // prefer mapped names if present
+                    // remove mapping not in order not to reuse for the next occurrence of the ref
+                    List<String> names = mappedNamesByRef.get(schema.get$ref());
+                    String mappedName = names.remove(0);
+                    anyOf.setBaseName(mappedName);
+                    anyOf.setName(toModelName(mappedName));
+                } else if (!org.apache.commons.lang3.StringUtils.isEmpty(schema.get$ref())) {
+                    // use $ref if it's reference
+                    String refName = ModelUtils.getSimpleRef(schema.get$ref());
+                    if (refName != null) {
+                        String modelName = toModelName(refName);
+                        anyOf.setName(modelName);
+                        anyOf.setBaseName(refName);
+                    }
+                } else if (anyOf.isArray) {
+                    // If the type is an array, extend the name with the inner type to prevent name collisions
+                    // in case multiple arrays with different types are defined. If the user has manually specified
+                    // a name, use that name instead.
+                    String collectionWithTypeName = toModelName(schema.getType()) + anyOf.containerTypeMapped + anyOf.items.dataType;
+                    String anyOfName = Optional.ofNullable(schema.getTitle()).orElse(collectionWithTypeName);
+                    anyOf.setName(anyOfName);
+                }
+                else {
+                    // In-placed type (primitive), because there is no mapping or ref for it.
+                    // use camelized `title` if present, otherwise use `type`
+                    String anyOfName = Optional.ofNullable(schema.getTitle()).orElseGet(schema::getType);
+                    anyOf.setName(toModelName(anyOfName));
+                }
+            }
+
+            // Set anyOf as oneOf for template processing since we want the same output
+            mdl.getComposedSchemas().setOneOf(newAnyOfs);
```

Comment:
this is an "interesting approach" as it tries to "reuse" oneOf for anyOF.

it may cause confusion to the end users reviewing the mustache templates (and they won't find any `anyOf` mustache tags) but I think we can give it a try to start with

cc 
@frol (2017/07) @farcaller (2017/08) @richardwhiuk (2019/07) @paladinzh (2020/05) @jacob-pro (2022/10) @dsteeley (2025/07)